server/handlers.go

```diff
@@ -211,37 +213,35 @@ func (s *Server) handleConnectorLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	authReqID := r.FormValue("req")
-
-	authReq, err := s.storage.GetAuthRequest(authReqID)
-	if err != nil {
-		s.logger.Errorf("Failed to get auth request: %v", err)
-		if err == storage.ErrNotFound {
-			s.renderError(r, w, http.StatusBadRequest, "Login session expired.")
-		} else {
-			s.renderError(r, w, http.StatusInternalServerError, "Database error.")
-		}
+	// Set the connector being used for the login.
+	if authReq.ConnectorID != "" && authReq.ConnectorID != connID {
```

Comment:
This check is here only because we pass connector id twice? Do we still need the ConnectorID field in the AuthRequest?

Reply:
Hmmm. I think it was here before because the auth request was already created, so we needed to check that it matched what we'd already agreed on. You're right that now it really only exists because we pass the connector ID twice to the login method (once as "connector_id" and once as "connector"), though one of those is specified by the client application, while the other comes from the choose-a-login-method page. I don't think it hurts to check, though I don't see how the two could be inconsistent at this point.

I think it's still useful to store the connector ID in the AuthRequest, because we also want to check it in e.g. handleConnectorCallback()